_posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title: "Generate mini site with Metanorma CLI"
+categories: about
+authors:
+  -
+    name: Abu Nashir
+    email: abunashir@gmail.com
+    social_links:
+      - https://github.com/abunashir
+
+excerpt: >-
+    Generate a collection mini site using Metanorma CLI
+---
+
+Each metanorma collection can be published as a mini site, there are couple of
+steps involed in this process to take a collection, compile it and generate the
+publishable site.
+
+To make this process easier, Metanorma has been offering a
+https://github.com/metanorma/mn-samples-itu/blob/master/Makefile[Makefile] that
+provides some targets to build the mini site. This has been working great until
+recently, but overtime it grew a lot and became harder to maintian, so we have
+decided to ship this as part of the CLI now.
+
+Let's go through a quick demo to see it in action and also customize the build
+
+=== Site Generation
+
+==== Demo Setup
+
+Let's take this https://github.com/metanorma/mn-samples-itu[ITU
+sample repo] as an exmaple, There are quite some files there but to keep the
+demo short let's pay attension to the following files and directory.
+
+[source,sh]
+----
+mn-samples-itu                          # Root directory
+|
+|-- sources                             # Source directory
+|   |-- images                          # Nested images for assets
+|   |-- G.191.adoc                      # Collection file inside sources
+|   |-- G.650.1-201803.adoc
+|
+|-- Gemfile                             # Gemfile in root directory
+|-- metanorma.yml                       # Configuration file in root directory
+----
+
+In root level, we have the `Gemfile`, `metanorma.yml` and `sources` directory.
+Inside `sources` there are two source files and an `images` folder for assets.
+
+==== Generate the site
+
+How can we generate the site now? Well it's pretty simple, all you need is the
+latest version of the https://github.com/metanorma/metanorma-cli[Metanorma CLI],
```

Comment:
Give the explicit version number: https://github.com/metanorma/metanorma-cli/releases/tag/v1.4.0

Reply:
That's a good idea, but regarding the exact tag, I assume we would have some fixes coming up in the new releases and I wouldn't prefer to update the blog post every time, so should we point to - https://github.com/metanorma/metanorma-cli/releases instead?